src/common/strtol.cc

```diff
@@ -148,41 +148,47 @@ T strict_iec_cast(std::string_view str, std::string *err)
     unit = str.substr(u, str.length() - u);
     // we accept both old si prefixes as well as the proper iec prefixes
     // i.e. K, M, ... and Ki, Mi, ...
-    if (unit.back() == 'i') {
-      if (unit.front() == 'B') {
-        *err = "strict_iecstrtoll: illegal prefix \"Bi\"";
-        return 0;
-      }
-    }
     if (unit.length() > 2) {
       *err = "strict_iecstrtoll: illegal prefix (length > 2)";
       return 0;
     }
-    switch(unit.front()) {
-      case 'K':
-        m = 10;
-        break;
-      case 'M':
-        m = 20;
-        break;
-      case 'G':
-        m = 30;
-        break;
-      case 'T':
-        m = 40;
-        break;
-      case 'P':
-        m = 50;
-        break;
-      case 'E':
-        m = 60;
-        break;
-      case 'B':
-        break;
-      default:
-        *err = "strict_iecstrtoll: unit prefix not recognized";
-        return 0;
+    if ((unit.back() == 'i') || (unit.length() == 1)) {
+      if (unit.back() == 'i') {
+        if (unit.front() == 'B') {
+          *err = "strict_iecstrtoll: illegal prefix \"Bi\"";
+          return 0;
+        }
+      }
```

Comment:
IMO we need to also catch cases like `BB` or `GG`. In total three cases to handle: 
1) `i` 
2) `Bi`
3) cases like `GG` or `TT`

I have some suggestions, feel free to amend/reject/modify:

```
// catch case for prefix "i"
if (unit.length() == 1) {
	if ((unit.front() == 'i') && (unit.back() == 'i')) {
		*err = "strict_iecstrtoll: illegal prefix \"i\"";
	}
	return 0;
}

// catch case "Bi" or repetitiveness like "GG" or "TT"
if (unit.length() == 2) {
	if (unit.front() == unit.back()) {
		*err = "strict_iecstrtoll: illegal prefix "  << "\"" << unit << "\"";
	} else if ((unit.front() == 'B') && (unit.back() == 'i')) { 
		*err = "strict_iecstrtoll: illegal prefix \"Bi\"";
	}
	if (!err.empty()) {
		return 0;
	}
}
```

Reply:
NVM, the switch is wrapped inside `if ((unit.back() == 'i') || (unit.length() == 1)) {`